lib/Parse/ParseDecl.cpp

```diff
@@ -2986,9 +2986,26 @@ ParserStatus Parser::parseNewDeclAttribute(DeclAttributes &Attributes,
     if (!Tok.is(tok::l_paren)) {
       // Normal access control attribute.
       AttrRange = Loc;
-      DuplicateAttribute = Attributes.getAttribute<AccessControlAttr>();
-      if (!DuplicateAttribute)
+      
+      if (!DiscardAttribute) {
         Attributes.add(new (Context) AccessControlAttr(AtLoc, Loc, access));
+        break;
+      }
+      
+      if (access != cast<AccessControlAttr>(*DuplicateAttribute).getAccess()) {
```

Comment:
Does this compile? [`SetterAccessAttr::classof`](https://github.com/apple/swift/blob/4f68365467fcee6a25e39a07243b1668bb4388b8/include/swift/AST/Attr.h#L1171-L1173) suggests the cast source should be a pointer.

Reply:
I see now, LLVM has a template to allow for calling `cast` with a reference. If the source is passed around by pointer, we prefer the template that takes and produces a pointer: `cast<AccessControlAttr>(DuplicateAttribute)`.

Reply:
It did compile, but your suggestion is more consistent with the rest of the codebase as you've shown me in our discussions. I improved it on the latest push I made.